src/dotnet/ExecutablePackageObtainer/ExecutablePackageObtainer.cs

```diff
@@ -0,0 +1,204 @@
+            string packageId,
+            string packageVersion = null,
+            FilePath nugetconfig = null,
+            string targetframework = null)
+        {
+            if (packageId == null) throw new ArgumentNullException(nameof(packageId));
+            if (targetframework == null)
+            {
+                targetframework = _bundledTargetFrameworkMoniker.Value;
+            }
+
+            var packageVersionOrPlaceHolder = new PackageVersion(packageVersion);
+
+            DirectoryPath individualToolVersion =
+                CreateIndividualToolVersionDirectory(packageId, packageVersionOrPlaceHolder);
+
+            FilePath tempProjectPath = CreateTempProject(
+                packageId,
+                packageVersionOrPlaceHolder,
+                targetframework,
+                individualToolVersion);
+
+            if (packageVersionOrPlaceHolder.IsPlaceHolder)
+            {
+                InvokeAddPackageRestore(
+                    nugetconfig,
+                    tempProjectPath,
+                    packageId);
+            }
+
+            InvokeRestore(nugetconfig, tempProjectPath, individualToolVersion);
+
+            if (packageVersionOrPlaceHolder.IsPlaceHolder)
+            {
+                var concreteVersion =
+                    new DirectoryInfo(
+                        Directory.GetDirectories(
+                            individualToolVersion.WithCombineFollowing(packageId).Value).Single()).Name;
+                DirectoryPath concreteVersionIndividualToolVersion =
+                    individualToolVersion.GetParentPath().WithCombineFollowing(concreteVersion);
+                Directory.Move(individualToolVersion.Value, concreteVersionIndividualToolVersion.Value);
+
+                individualToolVersion = concreteVersionIndividualToolVersion;
+                packageVersion = concreteVersion;
+            }
+
+            ToolConfiguration toolConfiguration = GetConfiguration(packageId, packageVersion, individualToolVersion);
+
+            return new ToolConfigurationAndExecutableDirectory(
+                toolConfiguration,
+                individualToolVersion.WithCombineFollowing(
+                    packageId,
+                    packageVersion,
+                    "tools",
+                    targetframework));
+        }
+
+        private static ToolConfiguration GetConfiguration(
+            string packageId,
+            string packageVersion,
+            DirectoryPath individualToolVersion)
+        {
+            FilePath toolConfigurationPath =
+                individualToolVersion
+                    .WithCombineFollowing(packageId, packageVersion, "tools")
+                    .CreateFilePathWithCombineFollowing("DotnetToolsConfig.xml");
+
+            ToolConfiguration toolConfiguration =
+                ToolConfigurationDeserializer.Deserialize(toolConfigurationPath.Value);
+            return toolConfiguration;
```

Comment:
I'd prefer a newline between a multi line statement and the next statement, especially when the next one is return.

Reply:
fixed